### PR TITLE
Fix static initializer priority for namespaces in LTO builds

### LINKED ISSUE #6575: LTO builds fail on Linux
Basically all tests fail right now on e4e2842, build worked fine as far as I see it.

https://build.icinga.com/view/Icinga%202/job/icinga2-snapshot/job/rpm-centos-6-2test/arch=x86_64/215/console

```
++ sudo icinga2 daemon -C
[2018-08-26 05:27:08 -0400] information/cli: Icinga application loader (version: v2.9.1-127-ge4e2842)
[2018-08-26 05:27:08 -0400] information/cli: Loading configuration file(s).
../cli/daemonutility.cpp:150: assertion failed: systemNS->Get("ApplicationType", &vAppType)
Caught SIGABRT.
```

## Your Environment
* Version used (`icinga2 --version`): e4e2842

### PR DESCRIPTION
**Pull request by dnsmichi**
Tested with both `ICINGA2_LTO_BUILD=OFF/ON`.

fixes #6575

### CODE REVIEW

File: lib/cli/daemonutility.cpp [hunk 1]
```diff
@@ -21,10 +21,12 @@
 #include "base/utility.hpp"
 #include "base/logger.hpp"
 #include "base/application.hpp"
+#include "base/serializer.hpp"
+#include "base/scriptglobal.hpp"
 #include "config/configcompiler.hpp"
 #include "config/configcompilercontext.hpp"
 #include "config/configitembuilder.hpp"
-
+#include <iostream>
```
Comment (Crunsher) [inline lib/cli/daemonutility.cpp]: Looks like you forgot to remove some debug changes :P

File: lib/cli/daemonutility.cpp [hunk 2]
```diff
@@ -145,7 +147,14 @@ bool DaemonUtility::ValidateConfigFiles(const std::vector<std::string>& configs,
 	if (!success)
 		return false;
 
+#ifdef I2_DEBUG
+	//std::cout << "Global NS: " << Serialize(ScriptGlobal::GetGlobals()) << std::endl;
+#endif /* I2_DEBUG */
```
Comment (Crunsher) [inline lib/cli/daemonutility.cpp]: Also here

File: lib/base/scriptframe.cpp [hunk 3]
```diff
@@ -51,7 +55,7 @@ INITIALIZE_ONCE_WITH_PRIORITY([]() {
 
 	Namespace::Ptr internalNS = new Namespace(l_InternalNSBehavior);
 	globalNS->SetAttribute("Internal", std::make_shared<ConstEmbeddedNamespaceValue>(internalNS));
-}, 50);
+}, 1000); /* This. */
```
Comment (Crunsher) [inline lib/base/scriptframe.cpp]: What's the reason for this comment?

File: lib/cli/daemonutility.cpp (hunk 2 above)
Comment (dnsmichi) [reply lib/cli/daemonutility.cpp]: That's on purpose, also the above include.

File: lib/base/scriptframe.cpp (hunk 3 above)
Comment (dnsmichi) [reply lib/base/scriptframe.cpp]: The comment above which explains the priority. I fear that this is changed or overseen in the future.

File: lib/cli/daemonutility.cpp (hunk 2 above)
Comment (Crunsher) [reply lib/cli/daemonutility.cpp]: It's dead code :skull:

**Comment by Crunsher** [comment]
Tested with Debian Jessie and gcc 8.2.0. With LTO and without, with LTO Icinga 2 does not terminate properly:
```
icinga2: /usr/include/boost/thread/pthread/recursive_mutex.hpp:113: void boost::recursive_mutex::lock(): Assertion `!pthread_mutex_lock(&m)' failed.
[1]    11140 abort      icinga2 daemon
```

This happens when run in the foreground as well as when started/stopped as a service. I'll rebuild the master with LTO enabled to find out whether this is a problem with this patch or with LTO builds themself.

**Comment by dnsmichi** [comment]
That's a known problem with debug builds, you won't see that with release builds. I'll drop the dead code for now, just for you :-P

**Pull request merged**